src/Contract/Transaction.purs

```diff
@@ -0,0 +1,153 @@
+-- | A module that defines the different transaction data types, balancing
+-- | functionality, signing and submission.
+module Contract.Transaction
+  ( balanceTx
+  , balanceTxM
+  , signTransaction
+  , submitTransaction
+  , module BalanceTxError
+  , module JsonWsp
+  , module ScriptLookups
+  , module Transaction
+  , module TxOutput
+  , module UnbalancedTx
+  ) where
+
+import Prelude
+import BalanceTx (balanceTx) as BalanceTx
+import BalanceTx (BalanceTxError) as BalanceTxError
+import Contract.Monad (Contract)
+import Data.Either (Either, hush)
+import Data.Maybe (Maybe)
+import Data.Newtype (wrap)
+import QueryM (signTransaction, submitTransaction) as QueryM
+import Types.JsonWsp (OgmiosTxOut, OgmiosTxOutRef) as JsonWsp --  Perhaps it's best not to expose `JsonWsp`.
+import Types.ScriptLookups
+  ( MkUnbalancedTxError(..) -- A lot errors so will refrain from explicit names.
+  , mkUnbalancedTx
+  ) as ScriptLookups
+import Types.Transaction (Transaction, TransactionHash)
+import Types.Transaction -- Most re-exported, don't re-export `Redeemer` and associated lens.
+  ( AuxiliaryData(AuxiliaryData)
+  , AuxiliaryDataHash(AuxiliaryDataHash)
+  , BootstrapWitness
+  , Certificate
+      ( StakeRegistration
+      , StakeDeregistration
+      , StakeDelegation
+      , PoolRegistration
+      , PoolRetirement
+      , GenesisKeyDelegation
+      , MoveInstantaneousRewardsCert
+      )
+  , CostModel(CostModel)
+  , Costmdls(Costmdls)
+  -- , DataHash(DataHash)
```

Comment:
Why is this commented out?

Reply:
I tried to keep things somewhat modular (although not always admittedly), so the commented out things are address or data related. Since it's part of the transaction, I can add them back in.